dev/Controller/SPController.php

```diff
@@ -42,28 +40,15 @@
  */
 final class SPController extends AbstractController
 {
-    private $identityProvider;
-    private $serviceProvider;
-    private $postBinding;
-
     public function __construct(
-        ServiceProvider $serviceProvider,
-        IdentityProvider $identityProvider,
-        PostBinding $postBinding
+        private readonly ServiceProvider $serviceProvider,
+        private readonly IdentityProvider $identityProvider,
+        private readonly PostBinding $postBinding
     ) {
-        $this->identityProvider = $identityProvider;
-        $this->serviceProvider = $serviceProvider;
-        $this->postBinding = $postBinding;
     }
 
-    /**
-     * @Route("/demo/sp", name="sp_demo")
-     *
-     * See @see RegistrationService for a more clean example.
-     *
-     * @throws \Exception
-     */
-    public function demoSpAction(Request $request)
+    #[Route(path: '/demo/sp', name: 'sp_demo')]
```

Comment:
ook attrubute gebruikt, top

Reply:
Took some getting used to, but in the end I much prefer the new attribute based annotations. More YAMLy FTW